### ISSUE TITLE
More testing

### ISSUE THREAD

**Issue by aapeliv** [Couchers-org/couchers#900]
We could do with a few more tests in the backend. There's some bits that are somewhat poorly tested. It'd be great to have 100% coverage on e.g. the auth code. (Password based login is not tested at all!)

**Comment by ShmuelTreiger** [comment]
It would be good if you could specify anywhere else you see that's missing. You mentioned auth, but any others? Or should we just use this thread to collect as we find untested things?

**Comment by lucaslcode** [comment]
https://develop--bcov.preview.couchershq.org

**Comment by aapeliv** [comment]
I've compiled a list of stuff that needs added tests from the coverage report. In addition to these, there's a whole bunch of error checking branches that I don't think need testing that badly, but I mean, it wouldn't hurt if someone is bored :P

## High priority

```
src/couchers/crypto.py: everything
src/couchers/interceptors.py: auth interceptors, other stuff is lower priority
src/couchers/servicers/auth.py: everything
src/couchers/utils.py: everything
```

## Files that seem to contain fully untested RPCs

```
src/couchers/servicers/communities.py: ListPlaces, ListGuides, ListEvents
src/couchers/servicers/groups.py: ListPlaces, ListGuides, ListEvents
src/couchers/servicers/conversations.py: GetUpdates
src/couchers/servicers/gis.py: currently nothing is tested
src/couchers/servicers/search.py: UserSearch needs more testing
src/couchers/servicers/requests.py: MarkLastSeenHostRequest?
src/couchers/tasks.py: send_password_changed_email?
```

I think `gis.py` doesn't have tests because I knew it was a temporary solution, needs a rewrite as per #897.

**Comment by aapeliv** [comment]
Friend requests are also completely untested

**Comment by trevvorko** [comment]
If no one has already worked on this I'd like to contribute. I could probably write tests for the high priority section detailed above. Let me know if there's anything I should be aware of before starting.

**Comment by aapeliv** [comment]
@ShmuelTreiger was going to work on this after some other issues, but I think it's fine for you to start working on this as I don't think Sam started yet. The tests are also quite decoupled so I'm sure there's also work for both of you if you wish to share!

**Comment by ShmuelTreiger** [comment]
By all means. It's on my todo list, but feel free!

**Comment by ShmuelTreiger** [comment]
Another area for improved testing would be ensuring that any api request which has a user_id:
1) filters for existing users (e.g: https://github.com/Couchers-org/couchers/blob/a0c66b6e058d6e1d1656a9a1cea5e4b3149ff708/app/backend/src/couchers/servicers/api.py#L433)

2) tests the api with a fake user (e.g. https://github.com/Couchers-org/couchers/blob/a0c66b6e058d6e1d1656a9a1cea5e4b3149ff708/app/backend/src/tests/test_requests.py#L36-L43)

**Comment by aapeliv** [comment]
Just a heads up to anyone reading this in the future, `session.query(User).filter(User.id == context.user_id).one()` is guaranteed by the auth interceptor to return a valid user and we don't need to check for existence of `context.user_id` (in "secure" servicers).

**Comment by kalvdans** [comment]
> src/couchers/servicers/search.py: UserSearch needs more testing

Has got its own bug now: https://github.com/Couchers-org/couchers/issues/1465

**Comment by aapeliv** [comment]
We're at 96%, which is pretty good.
